### ISSUE TITLE
DisallowedLooseComparisonRule should exclude DateTime(Immutable)

### ISSUE THREAD

**Issue by karstennilsen** [phpstan/phpstan-strict-rules#252]
```php
(new \DateTimeImmutable('2024-05-02 00:00:00')) === (new \DateTimeImmutable('2024-05-02 00:00:00'))
// returns false
```

Strict comparison makes not a lot of sense for DateTime objects in PHP. I like the rest to be strict in comparison, but not this.

**Comment by ondrejmirtes** [comment]
Duplicate of https://github.com/phpstan/phpstan-strict-rules/issues/232
